AQCU-1179: handle values of zero in field visit measurements

kmschoep-usgs opened the pull request:
Zeros in field visit measurements would break the log axis
functionality.  Fix was two-fold: added a check for zeros in field visit
measurement data when excludeZeroNegative was false or missing. Then
stripped zeros/negatives from field visit measurement data when
excludeZeroNegative was true.

Commit:
AQCU-1179: handle values of zero in field visit measurements

Zeros in field visit measurements would break the log axis
functionality.  Fix was two-fold: added a check for zeros in field visit
measurement data when excludeZeroNegative was false or missing. Then
stripped zeros/negatives from field visit measurement data when
excludeZeroNegative was true.

Commit:
code cleanup

lindsayplatt commented on R/dvhydrograph-render.R:
```diff
@@ -89,6 +97,11 @@ createDVHydrographPlot <- function(reportObject){
+  #remove zeros from field measurements, if present:
+  if (excludeZeroNegativeFlag && !isEmptyOrBlank(fieldVisitMeasurements)) {
+    fieldVisitMeasurements <- removeZeroNegative(fieldVisitMeasurements)
+  }
```
I think you only need to check `isLogged` in the if statement since `excludeZeroNegativeFlag` and `!isEmptyOrBlank(fieldVisitMeasurements)` were checked in order to create `logAxis`.

kmschoep-usgs replied:
This actually removes the zeros from the data if excludeZeroNegativeFlag = true; otherwise that doesn't happen and the report doesn't render if excludeZeroNegativeFlag = true and the zeros and negatives are not removed.

EDIT:  Ohhhh, I think I know what you mean now.  duh.  Yes.

lindsayplatt replied:
Right, but `isLogged` will only return true if it won't cause an error, and that function checks `excludeZeroNegativeFlag` inside. So this `if` can just become 

```
if (logAxis) {
    fieldVisitMeasurements <- removeZeroNegative(fieldVisitMeasurements)
}
```

lindsayplatt replied:
if there aren't zero's, this won't do anything.
if there are zero's, but `excludeZeroNegativeFlag = FALSE`, then `logAxis` will be FALSE
if there are zero's and `excludeZeroNegativeFlag = TRUE`, then `logAxis` will be TRUE

kmschoep-usgs replied:
Yep, gotcha-- that works!

lindsayplatt commented on R/dvhydrograph-render.R:
```diff
@@ -74,7 +74,15 @@ createDVHydrographPlot <- function(reportObject){
+  }
+  else {
+    fieldVisitMeasurementsCanLog <- TRUE
+  }
```
I think lines 78-83 should go inside of `parseFieldVisitMeasurements()` much like they do for `parseMinMaxIVs()`.

Commit:
AQCU-1179: simplify if statement when stripping zeros/negatives from field visit measurements

Commit:
AQCU-1179: handle the log-scalability flag for field visit measurements in the parse package, change how the zeroes/negative values are handled.

zmoore-usgs reviewed (approved):
Changes you made from Lindsay's feedback look good to me, but you might want to wait for @lindsaycarr to look at them as well.

lindsayplatt commented on R/utils-parse.r:
```diff
@@ -157,6 +158,18 @@ parseFieldVisitMeasurements <- function(reportObject){
+    }
+    else if(!isEmptyOrBlank(meas_Q[['value']]) && meas_Q[['value']] > 0 ) {
+      meas_Q[['canLog']] <- TRUE
+    }
```
kind of wondering if `meas_Q[['canLog']] <- isLogged(meas_Q, isVolFlow=TRUE, excludeZeroNegativeFlag)` can be used in lieu of all these if-elses (same change could be used in `parseMinMaxIVs`). Probably out of scope, so approving!
